go/vt/vtgate/scatter_conn.go

```diff
@@ -183,7 +183,14 @@ func (stc *ScatterConn) ExecuteMultiShard(
 
 			qs, err = getQueryService(rs, info)
 			if err != nil {
-				return nil, err
+				if info.reservedID == 0 || info.transactionID != 0 {
+					return nil, err
+				}
+				err = session.ResetShard(info.alias)
+				if err != nil {
+					return nil, err
+				}
+				qs = rs.Gateway
```

Comment:
Is the idea here that we route it through the gateway again and it will pick a tablet for us?
That's clever 💯